src/shims/unix/socket.rs

```diff
@@ -1,15 +1,24 @@
+use std::cell::RefCell;
+use std::collections::VecDeque;
 use std::io;
+use std::rc::Rc;
 
 use crate::shims::unix::*;
 use crate::*;
 
 use self::fd::FileDescriptor;
 
+//TODO: add comment
+const MAX_SOCKETPAIR_BUFFER_CAPACITY: usize = 212992;
+
 /// Pair of connected sockets.
-///
-/// We currently don't allow sending any data through this pair, so this can be just a dummy.
 #[derive(Debug)]
-struct SocketPair;
+struct SocketPair {
+    readbuf: Rc<RefCell<VecDeque<u8>>>,
+    writebuf: Rc<RefCell<VecDeque<u8>>>,
+    is_nonblock: bool,
+    _is_cloexec: bool,
```

Comment:
There's no need for this field since we don't support `exec`. Elsewhere in Miri we discard CLOEXEC flags entirely so I think we should do the same here.